src/main/java/org/tc/controllers/CourseController.java

```diff
@@ -32,35 +43,45 @@
     private CourseConverter courseConverter;
     @Autowired
     private CourseDTOConverter courseDTOConverter;
+    @Autowired
+    private UserCourseServiceInterface userCourseService;
+    @Autowired
+    private EvaluationServiceInterface evaluationService;
+
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        DefaultMessageCodesResolver defaultMessageCodesResolver = new DefaultMessageCodesResolver();
+        binder.setMessageCodesResolver(defaultMessageCodesResolver);
+    }
 
     @RequestMapping(value = {"/courses"}, method = RequestMethod.GET)
     public ModelAndView index() {
         ModelAndView mav = new ModelAndView("classpath:views/index");
-        mav.addObject("h1","Courses");
-        mav.addObject("courses", courseDTOConverter
-                .convertAll(courseService.getAll()));
+        mav.addObject("h1", "Courses");
+        List<Course> course = courseService.getAll();
+        List<CourseDTO> courses = courseDTOConverter
+                .convertAll(courseService.getAll());
+        mav.addObject("courses", courses);
         return mav;
     }
 
     @RequestMapping(value = {"/courses/{id}"}, method = RequestMethod.GET)
     public ModelAndView details(@PathVariable("id") int id) {
         Course course = courseService.getById(id);
-        if (course != null) {
-            ModelAndView mav = new ModelAndView("classpath:views/details");
-            mav.addObject("h1","Course details");
-            mav.addObject("course", course);
-            return mav;
-        } else {
-            throw new CourseNotFoundException("Update course");
-        }
+        ModelAndView mav = new ModelAndView("classpath:views/details");
+        mav.addObject("h1", "Course details");
```

Comment:
Because I have th:fragment header, where I set this value. On course page it's "Courses", on /details/, as here, it's "Course details" and so on